packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -1,4 +1,5 @@
 import React from 'react'
+import deepEqual from 'dequal'
```

Comment:
This has very few downloads/month. Is there a more popular one? Can we use one from the lodash suite for example?

Reply:
@mcollina I chose it because it does what it has to do, plus:

- It's well tested.
- It's tiny ([347B](https://bundlephobia.com/result?p=dequal@1.0.0) vs [3.8kB](https://bundlephobia.com/result?p=lodash.isequal@4.5.0) lodash's `isEqual`).
- It's [faster](https://github.com/lukeed/dequal#benchmarks) than lodash's `isEqual` (3.54x faster, to be specific).

Reply:
Awesome!

Reply:
I noticed that `dequal` doesn't compare functions:
```jsx
const dequal = require('dequal')
dequal((a, b) => a + b,  (a, b) => a + b) // false
```
So using passing `fetchOptionsOverrides` and `updateData` to `options` will break memoization.

Reply:
@lynxtaa

But neither `dequal` nor another package is able to do that kind of function comparison. In the following example the memoization will happen properly using `useCallback`:

```js
const { useCallback } from 'react'
const updateData = useCallback((old, new) => {
  return new
}, [])

const { data, refetch } = useQuery(GQL, { updateData })
```